src/d3a/models/area/redis_external_connection.py

```diff
@@ -0,0 +1,57 @@
+from redis import StrictRedis
+import json
+from d3a.d3a_core.redis_communication import REDIS_URL
+from d3a.models.strategy.external_strategy import ExternalStrategy
+
+
+class RedisExternalConnection:
+    def __init__(self, area):
+        self.area = area
+        self.redis_db = StrictRedis.from_url(REDIS_URL)
+        self.pubsub = self.redis_db.pubsub()
+        self.sub_to_area_event()
+        self.areas_to_register = []
+
+    def register_new_areas(self):
+        if not self.areas_to_register:
+            return
+
+        for new_area in self.areas_to_register:
+            area_object = self.area.__class__(name=new_area)
```

Comment:
I guess here you are instantiating the user defined area class? Correct me if Im wrong.

Reply:
Exactly. The reason I did not use the 'normal' way of constructing an are object, is circular references, if I imported Area class in this file this would result in circular import references.

Reply:
Wouldn'T it be a good approach to implement `abc` module to define minimum requirements for user defined classes?

Reply:
Actually for the strategies, there are none. The 'interface' is defined in the BaseStrategy class, which provides default implementations for the necessary methods for every strategy. Therefore, even if a new strategy just inherits from the BaseStrategy class, it is going to have the minimum functionality of it. That's why the ExternalStrategy class is so small, because the BaseStrategy provides default implementations for every required method.